File: hto/03/hto_03.c
#include <stdio.h>
#include <string.h>

int findRepeatNumber(int* nums, int numsSize) {
    int arr[numsSize];
    memset(arr, 0, sizeof(arr));
    for (int i = 0; i < numsSize; ++i) {
        if (arr[nums[i]] > 0) return nums[i];
        else                  arr[nums[i]]++;
    }
    return -1;
}

int main(void) {
    int a[] = {2, 3, 1, 0, 2, 5, 3};
    printf("%d\n", findRepeatNumber(a, sizeof(a) / sizeof(int)));
}

File: hto/03/hto_03.go
package main

import "fmt"

func findRepeatNumber(nums []int) int {
    for i := 0; i < len(nums); i++ {
        for nums[i] != i {
            if nums[i] == nums[nums[i]] {
                return nums[i]
            }
            nums[i], nums[nums[i]] = nums[nums[i]], nums[i]
        }
    }
    return -1
}

func main() {
	nums := []int{2, 3, 1, 0, 2, 5, 3}
	fmt.Println(findRepeatNumber(nums))
}
